src/s3/Cargo.toml

```diff
@@ -6,6 +6,7 @@ authors = ["Alex Crichton <alex@alexcrichton.com>"]
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/rust-lang/crates.io"
 description = "Interaction between crates.io and S3 for storing crate files"
+edition = "2018"
```

Comment:
I wasn't expecting a conversion to the 2018 edition in this PR, but it's not very much, so I'll allow it ;)